engine/src/test/java/com/google/android/fhir/sync/SyncJobTest.kt

```diff
@@ -198,4 +199,53 @@ class SyncJobTest {
 
     job.cancel()
   }
+
+  @Test
+  fun `sync time should update on every sync call`() = runBlockingTest {
+    val worker1 = PeriodicWorkRequestBuilder<TestSyncWorker>(15, TimeUnit.MINUTES).build()
+
+    // get flows return by work manager wrapper
+    val stateFlow1 = syncJob.stateFlow()
+
+    val stateList1 = mutableListOf<State>()
+
+    // convert flows to list to assert later
+    val job1 = launch { stateFlow1.toList(stateList1) }
+
+    val curentTimeStamp: OffsetDateTime = OffsetDateTime.now()
+    workManager
+      .enqueueUniquePeriodicWork(
+        SyncWorkType.DOWNLOAD_UPLOAD.workerName,
+        ExistingPeriodicWorkPolicy.REPLACE,
+        worker1
+      )
+      .result
+      .get()
+    Thread.sleep(5000)
+    val firstSyncResult = (stateList1[stateList1.size - 1] as State.Finished).result
+    assert(firstSyncResult.timestamp.compareTo(curentTimeStamp) > 0)
+    datastoreUtil.readLastSyncTimestamp()?.let { assert((it).compareTo(curentTimeStamp) > 0) }
```

Comment:
because this is testing a scenario, we don't really need to do `?.`. if anything we can just be more aggressive and use `!!`

```suggestion
    val timestamp = datastoreUtil.readLastSyncTimestamp()!!
    assertThat(timestamp).isEqualTo(...)
```

Reply:
Done